client/web/src/enterprise/insights/components/popover/Popover.tsx

```diff
@@ -0,0 +1,96 @@
+import ReachPopover, { Position, positionDefault } from '@reach/popover'
+import classnames from 'classnames'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import FocusLock from 'react-focus-lock'
+
+import { useKeyboard } from './hooks/use-keyboard'
+import { useOnClickOutside } from './hooks/use-outside-click'
+
+interface PopoverProps {
+    target: React.RefObject<HTMLElement>
+    position?: Position
+    open?: boolean
+    onVisibilityChange?: (open: boolean) => void
+    className?: string
+}
+
+export const Popover: React.FunctionComponent<PopoverProps> = props => {
+    const { target, position = positionDefault, children, open, onVisibilityChange, className } = props
+
+    const isControlled = useRef(open !== undefined)
```

Comment:
We need to allocate time for custom ESLint rules to avoid annoying comments 😬

```suggestion
    const isControlledReference = useRef(open !== undefined)
```